packages/vite/src/node/config.ts

```diff
@@ -242,14 +242,14 @@ export async function resolveConfig(
 
   // run config hooks
   const userPlugins = [...prePlugins, ...normalPlugins, ...postPlugins]
-  userPlugins.forEach((p) => {
+  for (const p of userPlugins) {
     if (p.config) {
-      const res = p.config(config, configEnv)
+      const res = await p.config(config, configEnv)
```

Comment:
**thought (if-minor, perf):** Is it possible to use e.g. `Promise.all` (or similar) here to make all the awaits in the loop in parallel?